GOES/Scripts/alert_hrc.py

```diff
@@ -0,0 +1,182 @@
+    #Cast astropy table data into json serializable types
+    recent_data = {'time': str(time),
+                   'hrc_proxy': int(hrc_proxy),
+                   'hrc_proxy_legacy': int(hrc_proxy_legacy)}
+
+    #Check current status of HRC proxy violations.
+    #If one has been found very recently, do not email about the violation again
+    with open(VIOL_RECORD_FILE) as f:
+        curr_viol = json.load(f)
+    
+    #Iterate over kinds of threshold and versions each proxy
+    content = ''
+    for kind in HRC_THRESHOLD.keys():
+        for proxy in ['hrc_proxy', 'hrc_proxy_legacy']:
+            if recent_data[proxy] > HRC_THRESHOLD[kind]:
+                #check if there was a similar kind of violation withing the last 24 hours
+                if viol_time_check(curr_viol, kind, proxy):
+                    content += f"{kind}: {proxy}\n"
+                    content += f"Observed: {recent_data[proxy]:.5e} at Time: {time}\n"
+                    content += f"Limit: {HRC_THRESHOLD[kind]:.3e} counts/sec.\n"
+                    content += f"{'-' * 20}\n"
+                    curr_viol[f"{kind}_{proxy}"] = recent_data
+    
+    if content != '' and HRC_ADMIN != []:
+        send_mail(content, "HRC Proxy Violation", HRC_ADMIN)
+    with open(VIOL_RECORD_FILE, 'w') as f:
+        json.dump(curr_viol, f, indent = 4)
+    
+    add_to_archive(recent_data, HRC_PROXY_DATA_FILE)
+
+def send_mail(content, subject, admin):
+    """
+    send out a notification email to admin
+    """
+    content += f'This message was send to {" ".join(admin)}'
+    cmd = f'echo "{content}" | mailx -s "{subject}" {" ".join(admin)}'
+    os.system(cmd)
+
+def viol_time_check(curr_viol, kind, proxy):
+    """
+    Prevents spamming violation emails if the data is in violation, 
+    opting to send out a email if the specific violation was last warned more than 24 hours ago.
+    """
+    time_string = curr_viol[f"{kind}_{proxy}"]['time']
+    last = datetime.datetime.strptime(time_string, '%Y:%j:%H:%M')
+    now = datetime.datetime.utcnow()
+    return (now - last).days > 1    
+
+def add_to_archive(recent_data, outfile):
+    #data_line = f"{recent_data['time']:<13}\t{recent_data['hrc_proxy']:<9}\t{recent_data['hrc_proxy_legacy']:<16}\n"
+    data_line = f"{recent_data['time']}\t{recent_data['hrc_proxy']}\t{recent_data['hrc_proxy_legacy']}\n"
+    if os.path.isfile(outfile):
+        mode = 'a'
+        #Secondary check in appending to the archive in case there is a time discrepancy.
+        append_time = datetime.datetime.strptime(recent_data['time'], '%Y:%j:%H:%M')
+        out = subprocess.check_output(f"tail -n 1 {outfile}", shell=True, executable='/bin/csh').decode()
+        last_time = datetime.datetime.strptime(out.split()[0], '%Y:%j:%H:%M')
+        #Send alert if the archive has not been recording for 15 minutes
+        if (append_time - last_time).total_seconds() > 900:
```

Comment:
The check if the archive is updating with the expected cadence should be moved to an independent script to avoid a situation where the code breaks due to some issue before this check has a chance to be performed, which would prevent the notification in line 96 from being sent.

Reply:
Refactored into a check_archive.py script instead.